onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlSTFT.h

```diff
@@ -0,0 +1,610 @@
+{
+public:
+    STDMETHOD(CreateKernel)(
+        IMLOperatorKernelCreationContext* context,
+        IMLOperatorKernel** kernel)
+    {
+        try
+        {
+            auto dftOperator = wil::MakeOrThrow<DmlSTFTOperator>(context);
+            dftOperator.CopyTo(kernel);
+            return S_OK;
+        }
+        catch (...)
+        {
+            return E_FAIL;
+        }
+    }
+
+    static void RegisterSTFTKernel(IMLOperatorRegistry* registry)
+    {
+        MLOperatorKernelDescription kernelDescription = {};
+        kernelDescription.domain = "";
+        kernelDescription.name = "STFT";
+        kernelDescription.minimumOperatorSetVersion = 17;
+        kernelDescription.executionType = MLOperatorExecutionType::D3D12;
+
+        // T1: tensor(float16), tensor(float), tensor(double), tensor(bfloat16)
+        MLOperatorEdgeTypeConstrant t1Constraint;
+        t1Constraint.typeLabel = "T1";
+        std::array<MLOperatorEdgeDescription, 2> t1AllowedEdges
+        {
+            MLOperatorEdgeDescription { MLOperatorEdgeType::Tensor, (uint64_t)MLOperatorTensorDataType::Float16 },
+            MLOperatorEdgeDescription { MLOperatorEdgeType::Tensor, (uint64_t)MLOperatorTensorDataType::Float },
+        };
+        t1Constraint.allowedTypes = t1AllowedEdges.data();
+        t1Constraint.allowedTypeCount = static_cast<uint32_t>(t1AllowedEdges.size());
+
+        // T2 : tensor(int32), tensor(int64)
+        MLOperatorEdgeTypeConstrant t2Constraint;
+        t2Constraint.typeLabel = "T2";
+        std::array<MLOperatorEdgeDescription, 2> t2AllowedEdges
+        {
+            MLOperatorEdgeDescription { MLOperatorEdgeType::Tensor, (uint64_t)MLOperatorTensorDataType::Int32 },
+            MLOperatorEdgeDescription { MLOperatorEdgeType::Tensor, (uint64_t)MLOperatorTensorDataType::Int64 },
+        };
+        t2Constraint.allowedTypes = t2AllowedEdges.data();
+        t2Constraint.allowedTypeCount = static_cast<uint32_t>(t2AllowedEdges.size());
+
+        std::array<MLOperatorEdgeTypeConstrant, 2> typeConstraints{ t1Constraint, t2Constraint };
+        kernelDescription.typeConstraints = typeConstraints.data();
+        kernelDescription.typeConstraintCount = static_cast<uint32_t>(typeConstraints.size());
+
+        MLOperatorAttributeNameValue onesidedAttributeValue;
+        onesidedAttributeValue.name = "onesided";
+        onesidedAttributeValue.type = MLOperatorAttributeType::Int;
+        onesidedAttributeValue.valueCount = 1;
+        static const int64_t onesided[] = { 1 };
+        onesidedAttributeValue.ints = onesided;
+
+        std::array<MLOperatorAttributeNameValue, 1> attributeDefaultValues{ onesidedAttributeValue };
+
+        kernelDescription.defaultAttributes = attributeDefaultValues.data();
+        kernelDescription.defaultAttributeCount = static_cast<uint32_t>(attributeDefaultValues.size());
+        kernelDescription.options = MLOperatorKernelOptions::None;
+        kernelDescription.executionOptions = 0;
```

Comment:
>executionOptions = 0; [](http://example.com/codeflow?start=26&length=21)

Oh interesting, never noticed this field before.

`//! Reserved for additional options.  Must be zero.`